app/services/images.py

```diff
@@ -0,0 +1,83 @@
+import os
+from io import BytesIO
+from typing import List, Tuple
+from uuid import uuid4
+
+from fastapi import UploadFile
+from httpx import AsyncClient
+from PIL import Image
+
+from app.log import get_logger
+
+logger = get_logger("__name__")
+
+
+class ImageService:
+    def _setup_download_path(self, query: str) -> str:
+        """Makes sure that download path (based on query) exists, then returns it."""
+        base_path = self.get_downloaded_path()
+        path = f"{base_path}/{query}"
+        if not os.path.exists(path):
+            os.makedirs(path)
+        return path
+
+    def _setup_upload_path(self) -> str:
+        """Makes sure that path for uploading files exists."""
+        path = self.get_upload_path()
+        if not os.path.exists(path):
+            os.makedirs(path)
+        return path
+
+    async def download_urls(self, query: str, urls: List[str]) -> List[str]:
+        base_path = self._setup_download_path(query)
+        file_paths = []
+        async with AsyncClient() as client:
+            for url in urls:
+                uid = uuid4()
+                dest = f"{base_path}/{uid}.jpg"
```

Comment:
Images downloaded get a new name based on a uuid. I thought this would be cleanest, but it does not account for duplicates if you query the same thing a couple times back to back. I also just make assumptions from twitter search about the image size/type.